.github/workflows/ci.yaml

```diff
@@ -0,0 +1,48 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+  clone:
+    runs-on: self-hosted
```

Comment:
If you want, you could try run this on the GHA runner?
We have 3,000 Actions minutes/month available 😄